Move option setup for vt_trace_only before trace-only target is created

**What Needs to be Done?**
Currently the vt_trace_only option `option(vt_trace_only "Build VT with trace-only mode enabled" ON)` is added after `VT_TRACE_LIB` target is created:
![image](https://user-images.githubusercontent.com/9077677/107554283-0706ed80-6bd6-11eb-8fc9-d2e3c29fdffa.png)

This works fine as long as you add cmake flag `-Dvt_trace_only` to enable/disable trace-only target. Without the flag the 'vt_trace_only' option is set (by default) to ON and later CMake tries to use the non-existent target which results in CMake Error:
![image](https://user-images.githubusercontent.com/9077677/107555160-1f2b3c80-6bd7-11eb-8fe4-d62380720e10.png)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,6 +110,11 @@ include(cmake/load_bundled_libraries.cmake)
 
 include(cmake/nvcc_no_deprecated_gpu_targets.cmake)
 
+option(vt_trace_only "Build VT with trace-only mode enabled" ON)
+if (vt_trace_only)
+  message(STATUS "Building additional target for VT in trace-only mode")
+endif()
+
 # Primary VT build
 add_subdirectory(src)
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,7 @@ include(cmake/load_bundled_libraries.cmake)
 
 include(cmake/nvcc_no_deprecated_gpu_targets.cmake)
 
-option(vt_trace_only "Build VT with trace-only mode enabled" ON)
+option(vt_trace_only "Build VT with trace-only mode enabled" OFF)
 if (vt_trace_only)
   message(STATUS "Building additional target for VT in trace-only mode")
 endif()

--- a/cmake/define_build_types.cmake
+++ b/cmake/define_build_types.cmake
@@ -104,11 +104,6 @@ else()
   set(vt_feature_cmake_trace_enabled "0")
 endif()
 
-option(vt_trace_only "Build VT with trace-only mode enabled" ON)
-if (vt_trace_only)
-  message(STATUS "Building additional target for VT in trace-only mode")
-endif()
-
 # This will be set to 1 during generation of cmake config for vt-trace target
 set(vt_feature_cmake_trace_only "0")
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -252,13 +252,12 @@ link_target_with_vt(
   DEFAULT_LINK_SET
 )
 
-# Define and set up the different build types of VT
-
 include(../cmake/trace_only_functions.cmake)
 if (vt_trace_only)
   create_trace_only_target()
 endif()
 
+# Define and set up the different build types of VT
 include(../cmake/define_build_types.cmake)
 
 target_include_directories(

